src/conventions.jl

```diff
@@ -0,0 +1,33 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+Convention used to reconstruct the faces of a polytope
+from a list of vertices.
+"""
+abstract type OrderingConvention end
+
+"""
+    connectivity(polytope, convention, k)
+
+Return the list of `Connectivity` elements used to construct
+all `k`-faces of a polytope, following an ordering convention.
+"""
+function connectivity(polytope::Type{<:Polytope}, convention::OrderingConvention, k::Integer)
+  connectivity(polytope, convention, Val(k))
+end
+
+# ---------------------------
+# CONVENTION IMPLEMENTATIONS
+# ---------------------------
+
+include("conventions/gmsh.jl")
+
+function facets(p::Polytope{N}, ordering=GMSH) where {N}
+  faces(p, N-1, ordering)
+end
+
+function faces(p::Polytope{N,Dim}, rank, ordering=GMSH) where {N,Dim}
```

Comment:
Why request the type parameters `N,Dim` if they are not used in the implementation? Also, notice how you flip the `rank` argument from second argument in faces to 3 argument in connectivity. These details are important, and we want to strike a consistent API where users don't need to lookup signatures repeatedly.

Reply:
You're totally right. I really hope I'll see this kind of things more easily because even if I took care with this PR these inconsistencies passed through..

Reply:
I am happy that you are taking the suggestions gracefully, it will really help in the long run as we evolve the package together :) We need to push each other to achieve results that we wouldn't be able to achieve alone :100: